pyquil/quilbase.py

```diff
@@ -199,11 +199,23 @@ def out(self) -> str:
                 _format_qubits_out(self.qubits),
             )
 
-    def controlled(self, control_qubit: QubitDesignator) -> "Gate":
+    def controlled(self, control_qubits: Union[QubitDesignator, List[QubitDesignator]]) -> "Gate":
         """
-        Add the CONTROLLED modifier to the gate with the given control qubit.
+        Add the CONTROLLED modifier to the gate with the given control qubit or list of control
+        qubits.
         """
-        control_qubit = unpack_qubit(control_qubit)
+        if isinstance(control_qubits, list):
+            if len(control_qubits) == 0:
+                return self
+            else:
```

Comment:
nitpick: since the `if` block ends in a `return`, remove the `else:` and un-indent.

Reply:
I guess it won't be an issue, as I will change to my code to iterative version. However, note that in my code we distinguish list of qubits of single qubit not in a list. Does this require different behaviour parts of code?

Reply:
Removing the `else:` and un-indenting the else block here should be equivalent. To be clear, I mean that

```python
if len(control_qubits) == 0:
    return self
else:
    ...
```

should change to

```python
if len(control_qubits) == 0:
    return self
...
```

Reply:
Oh, I see I understood that I have to un-indent twice;)